src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -0,0 +1,26 @@
+package cromwell.engine.db
+
+import java.util.Date
+
+import cromwell.binding.Call
+import cromwell.engine.store.SymbolStoreEntry
+import cromwell.engine.{WorkflowId, WorkflowState}
+
+trait DataAccess {
+  // Will stamp the start_dt column with DateTime.Now.
+  // Will stamp the WorkflowState as Submitted.
+  def createWorkflow(id: WorkflowId, wdlUri: String, symbols: Seq[SymbolStoreEntry]): Unit
+
+  def updateWorkflow(id: WorkflowId, state: WorkflowState): Unit
+
+  def updateCall(id: WorkflowId, call: Call, callStatus: Option[CallStatus],
+                 callInfo: Option[CallInfo], symbols: Option[Seq[SymbolStoreEntry]]): Unit
+
+  def query(workflowId: Option[Seq[WorkflowId]] = None,
+            wdlUris: Option[Seq[String]] = None,
+            states: Option[Seq[WorkflowState]] = None,
+            beforeStart: Option[Date] = None,
+            afterStart: Option[Date] = None,
+            beforeEnd: Option[Date] = None,
+            afterEnd: Option[Date] = None): Seq[QueryWorkflowExecutionResult]
```

Comment:
I was imagining Slick as having something which would enforce that it *is* a Set (as opposed to de-duping). I just made that feature up though so perhaps it doesn't exist.